lib/segment/src/types.rs

```diff
@@ -99,6 +99,8 @@ impl FromStr for DateTimePayloadType {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // Attempt to parse the input string in RFC 3339 format
         if let Ok(datetime) = chrono::DateTime::parse_from_rfc3339(s)
```

Comment:
How about also using [`to_rfc3339`](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.to_rfc3339) at the time we convert it into a string?